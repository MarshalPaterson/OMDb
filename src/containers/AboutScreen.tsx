import React from 'react';
import {Component} from 'react';
import {Dimensions, StyleSheet, Text, View, Image} from 'react-native';

const {width, height} = Dimensions.get('window');

type Props = {};
export default class AboutScreen extends Component<Props> {
  render() {
    return (
      <View style={styles.container}>
        <View style={styles.bgImageWrapper}>
          <Image
            source={require('../assets/m.png')}
            style={styles.bgImage}
          />
        </View>
        <Text style={styles.welcome}>
          Welcome to Marvel Heroes, a React Native example.
        </Text>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
  },
  welcome: {
    fontSize: 18,
    marginTop: 320,
    textAlign: 'center',
    color: 'white',
  },
  instructions: {
    textAlign: 'center',
    color: 'white',
    marginBottom: 5,
  },
  bgImageWrapper: {
    position: 'absolute',
    top: 0,
    bottom: 0,
    left: 0,
    right: 0,
  },
  bgImage: {
    flex: 1,
    resizeMode: 'cover',
    width: width,
    height: height,
  },
});
