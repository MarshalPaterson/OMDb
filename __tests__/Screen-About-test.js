import React from 'react';
import renderer from 'react-test-renderer';
import AboutScreen from '../src/containers/AboutScreen';

jest.useFakeTimers();

test('renders correctly', async () => {
  const tree = renderer.create(<AboutScreen />).toJSON();
  expect(tree).toMatchSnapshot();
});