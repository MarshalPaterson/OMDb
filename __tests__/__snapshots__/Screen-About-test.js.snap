// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <Image
      source={
        Object {
          "testUri": "../../../src/assets/st.jpg",
        }
      }
      style={
        Object {
          "flex": 1,
          "height": 1334,
          "resizeMode": "cover",
          "width": 750,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 22,
        "fontWeight": "bold",
        "marginTop": 320,
        "textAlign": "center",
      }
    }
  >
    Welcome to Star Trek, a React Native example.
  </Text>
</View>
`;
